src/github.com/getlantern/flashlight/proxied/proxied.go

```diff
@@ -300,22 +365,13 @@ func httpClient(rootCA string, proxyAddrFN eventual.Getter, persistent bool) (*h
 		}
 	}
 
-	if proxyAddrFN != nil {
-		log.Debug("Waiting for proxy server to come online")
-		proxyAddr, ok := proxyAddrFN(60 * time.Second)
+	tr.Proxy = func(req *http.Request) (*url.URL, error) {
+		proxyAddr, ok := getProxyAddr()
 		if !ok {
-			// Instead of finishing here we just log the error and continue, the client
-			// we are going to create will surely fail when used and return errors,
-			// those errors should be handled by the code that depends on such client.
-			log.Errorf("Proxy never came online")
+			return nil, ErrChainedProxyUnavailable
```

Comment:
I think this  is a good change. If we want direct connection at some point we should be able to do it at the library user side. Just providing the proper error signal from the Chained/Fronted API should be fine.
Besides, it wasn't obvious that it was doing that (i.e. too "smart" behavior)